code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -511,6 +511,35 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
 
 			apply_overlay(GLASSES_LAYER)
 
+/mob/living/carbon/human/update_inv_glasses()
+	remove_overlay(GLASSES_LAYER)
+	remove_overlay(GOGGLES_LAYER)
+	if(glasses)
+		if(istype(glasses,/obj/item/clothing/glasses/hud/medgoggles))
+			if(client && hud_used &&  hud_used.hud_shown && hud_used.inventory_shown)
+				glasses.screen_loc = ui_glasses
+				client.screen += glasses
+			if(glasses.icon_override)
+				overlays_standing[GOGGLES_LAYER] = image("icon" = glasses.icon_override, "icon_state" = "[glasses.icon_state]", "layer" =-GOGGLES_LAYER)
+			else if(glasses.sprite_sheets && glasses.sprite_sheets[species.name])
+				overlays_standing[GOGGLES_LAYER]= image("icon" = glasses.sprite_sheets[species.name], "icon_state" = "[glasses.icon_state]", "layer" =-GOGGLES_LAYER)
+			else
+				overlays_standing[GOGGLES_LAYER]= image("icon" = 'icons/mob/eyes.dmi', "icon_state" = "[glasses.icon_state]", "layer" =-GOGGLES_LAYER)
+
+			apply_overlay(GOGGLES_LAYER)
+		else
+			if(client && hud_used &&  hud_used.hud_shown && hud_used.inventory_shown)
+				glasses.screen_loc = ui_glasses
+				client.screen += glasses
+			if(glasses.icon_override)
+				overlays_standing[GLASSES_LAYER] = image("icon" = glasses.icon_override, "icon_state" = "[glasses.icon_state]", "layer" =-GLASSES_LAYER)
+			else if(glasses.sprite_sheets && glasses.sprite_sheets[species.name])
+				overlays_standing[GLASSES_LAYER]= image("icon" = glasses.sprite_sheets[species.name], "icon_state" = "[glasses.icon_state]", "layer" =-GLASSES_LAYER)
+			else
+				overlays_standing[GLASSES_LAYER]= image("icon" = 'icons/mob/eyes.dmi', "icon_state" = "[glasses.icon_state]", "layer" =-GLASSES_LAYER)
+
+			apply_overlay(GLASSES_LAYER)
+
```

Comment:
duplicate code

Reply:
```
/mob/living/carbon/human/update_inv_glasses()
	remove_overlay(GLASSES_LAYER)
	remove_overlay(GOGGLES_LAYER)
	if(glasses)
		if(istype(glasses,/obj/item/clothing/glasses/mgoggles))
			if(client && hud_used &&  hud_used.hud_shown && hud_used.inventory_shown)
				glasses.screen_loc = ui_glasses
				client.screen += glasses
			if(glasses.icon_override)
				overlays_standing[GOGGLES_LAYER] = image("icon" = glasses.icon_override, "icon_state" = "[glasses.icon_state]", "layer" =-GOGGLES_LAYER)
			else if(glasses.sprite_sheets && glasses.sprite_sheets[species.name])
				overlays_standing[GOGGLES_LAYER]= image("icon" = glasses.sprite_sheets[species.name], "icon_state" = "[glasses.icon_state]", "layer" =-GOGGLES_LAYER)
			else
				overlays_standing[GOGGLES_LAYER]= image("icon" = 'icons/mob/eyes.dmi', "icon_state" = "[glasses.icon_state]", "layer" =-GOGGLES_LAYER)
			apply_overlay(GOGGLES_LAYER)
		else
			if(client && hud_used &&  hud_used.hud_shown && hud_used.inventory_shown)
				glasses.screen_loc = ui_glasses
				client.screen += glasses
			if(glasses.icon_override)
				overlays_standing[GLASSES_LAYER] = image("icon" = glasses.icon_override, "icon_state" = "[glasses.icon_state]", "layer" =-GLASSES_LAYER)
			else if(glasses.sprite_sheets && glasses.sprite_sheets[species.name])
				overlays_standing[GLASSES_LAYER]= image("icon" = glasses.sprite_sheets[species.name], "icon_state" = "[glasses.icon_state]", "layer" =-GLASSES_LAYER)
			else
				overlays_standing[GLASSES_LAYER]= image("icon" = 'icons/mob/eyes.dmi', "icon_state" = "[glasses.icon_state]", "layer" =-GLASSES_LAYER)

			apply_overlay(GLASSES_LAYER)`\


```
I think I need to make 
`if(istype(glasses,/obj/item/clothing/glasses/mgoggles))` 
include
 `obj/item/clothing/glasses/hud/medgoggles` 
to make it not do this
![scrnshot1](https://user-images.githubusercontent.com/75421103/101244535-1d279780-36d5-11eb-8ec3-f991a629344e.png)

Because this is neat.
![Goggles Work, front](https://user-images.githubusercontent.com/75421103/101244562-3d575680-36d5-11eb-88e2-65edd219946c.png)

I will figure out how to do it in a couple days when I'm off work, I'm trying to learn how to do this stuff so thanks for humoring it.